src/App.tsx

```diff
@@ -1,26 +1,134 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import * as todoService from './api/todos';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { FilterBy } from './types/FilterBy';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [valueError, setValueError] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.All);
+  const [isNotificationHidden, setIsNotificationHidden] = useState(true);
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const autoHideNotification = () => {
+    setTimeout(() => {
+      setIsNotificationHidden(true);
+    }, 3000);
+  };
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const todosFromServer = await todoService.getTodos();
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        setValueError('Unable to load todos');
+        setIsNotificationHidden(false);
```

Comment:
I think we can have only one state to control error,  so when the error is not empty string we show notification
` { hidden: !valueError.length }`